### PR TITLE
Implemented few more stubs for Content View functionality in API

### PR DESCRIPTION
**Pull request by oshtaier**
```
nosetests tests/foreman/api/test_contentview.py -m test_positive_promote_out_of_sequence
.
----------------------------------------------------------------------
Ran 1 test in 103.879s

OK
```
```
nosetests tests/foreman/api/test_contentview.py -m test_negative_update_label
S
----------------------------------------------------------------------
Ran 1 test in 9.693s

OK (SKIP=1)
```
```
nosetests tests/foreman/api/test_contentview.py -m test_positive_publish_rh
..
----------------------------------------------------------------------
Ran 2 tests in 128.481s

OK
```
```
nosetests tests/foreman/api/test_contentview.py -m test_positive_promote_rh
..
----------------------------------------------------------------------
Ran 2 tests in 172.624s

OK
```

### CODE REVIEW

File: tests/foreman/api/test_contentview.py [hunk 1]
```diff
@@ -343,7 +344,7 @@ def test_positive_publish_multiple(self):
         """@Test: Publish a content view several times.
 
         @Assert: Content view has the correct number of versions after each
-        promotion.
+        publishing operation.
```
Comment (elyezer) [inline tests/foreman/api/test_contentview.py]: :+1:

**Comment by elyezer** [comment]
ACK

File: tests/foreman/api/test_contentview.py [hunk 2]
```diff
@@ -802,6 +803,29 @@ def test_positive_promote_composite_multiple_content_multiple(self):
...
+
+        @Feature: ContentView
+        """
+        content_view = entities.ContentView(organization=self.org).create()
+        for _ in range(REPEAT):
+            content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), REPEAT)
+        self.assertEqual(len(content_view.version[0].read().environment), 0)
+        lce_list = content_view.version[REPEAT - 1].read().environment
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: are you trying to get the last item of the list? you can use list[-1]

File: tests/foreman/api/test_contentview.py [hunk 3]
```diff
@@ -802,6 +803,29 @@ def test_positive_promote_composite_multiple_content_multiple(self):
...
+        @Assert: Content view promoted out of sequence properly
+
+        @Feature: ContentView
+        """
+        content_view = entities.ContentView(organization=self.org).create()
+        for _ in range(REPEAT):
+            content_view.publish()
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version), REPEAT)
+        self.assertEqual(len(content_view.version[0].read().environment), 0)
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: Can you please explain what this statement does?

File: tests/foreman/api/test_contentview.py [hunk 4]
```diff
@@ -802,6 +803,29 @@ def test_positive_promote_composite_multiple_content_multiple(self):
...
+        lce_list = content_view.version[REPEAT - 1].read().environment
+        self.assertEqual(len(lce_list), 1)
+        promote(content_view.version[0], lce_list[0].id, force=True)
+        content_view = content_view.read()
+        self.assertEqual(len(content_view.version[0].read().environment), 1)
+        self.assertEqual(
+            len(content_view.version[REPEAT - 1].read().environment), 0)
+
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: Can you add some comments to these assert statements, it is difficult to understand what is being asserted without the comments.

File: tests/foreman/api/test_contentview.py [hunk 5]
```diff
@@ -973,205 +994,104 @@ def test_positive_add_rh_custom_spin(self):
...
-        # in filter)
-        #   * A filter on severity (only content of specific errata
-        # severity.
+    def test_positive_publish_rh(self):
+        """Attempt to publish a content view containing Red Hat content
 
-    # Content View: promotions
-    # katello content view promote --label=MyView --env=Dev --org=ACME
-    # katello content view promote --view=MyView --env=Staging --org=ACME
+        @Assert: Necessary content view can be published
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: remove `necessary`

File: tests/foreman/api/test_contentview.py [hunk 6]
```diff
@@ -973,205 +994,104 @@ def test_positive_add_rh_custom_spin(self):
...
-    @stubbed()
-    def test_positive_publish_rh(self):
-        """
-        @test: attempt to publish a content view containing RH content
-        @feature: Content Views
-        @setup: Multiple environments for an org; RH content synced
-        @assert: Content view can be published
-        """
-        # See method test_subscribe_system_to_cv in module test_contentview_v2
+        @Assert: Necessary content view can be promoted
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: remove `necessary`

File: tests/foreman/api/test_contentview.py [hunk 7]
```diff
@@ -973,205 +994,104 @@ def test_positive_add_rh_custom_spin(self):
...
-        gets updated
-        @feature: Content Views
-        @setup: Multiple environments for an org; multiple versions
-        of a content view created/published
-        @steps:
-        1. publish a view to an environment noting the CV version
-        2. edit and republish a new version of a CV
-        @assert: Content view version is updated in target environment.
-        @status: Manual
+        @Assert: Necessary content view can be promoted
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: Remove `Necessary`

**Comment by sthirugn** [comment]
Pending comments

File: tests/foreman/api/test_contentview.py (hunk 4 above)
Comment (sghai) [inline tests/foreman/api/test_contentview.py]: yes please. that helps to quickly understand the test flow.

**Comment by oshtaier** [comment]
Thanks for the comments. Fixed code according to them

File: tests/foreman/api/test_contentview.py [hunk 8]
```diff
@@ -804,6 +804,37 @@ def test_positive_promote_composite_multiple_content_multiple(self):
...
+        content_view = entities.ContentView(organization=self.org).create()
+        for _ in range(REPEAT):
+            content_view.publish()
+        content_view = content_view.read()
+        # Check that CV is published and has proper number of CV versions.
+        self.assertEqual(len(content_view.version), REPEAT)
+        # After each publish operation application re-assign environment to
+        # latest CV version. Correspondingly, at that moment, first cv version
+        # should have 0 environments and latest should have one ('Library')
+        # assigned to it.
```
Comment (sthirugn) [inline tests/foreman/api/test_contentview.py]: :+1:

**Comment by sthirugn** [comment]
ACK

**Comment by sthirugn** [comment]
This PR has two acks already.

**Pull request merged**